### ISSUE TITLE
Setup guide - wrong java JDK version linked.

### ISSUE THREAD

**Issue by mrdude2478** [borntohonk/Switch-Ghidra-Guides#9]
Thanks for posting you guide, although you made a slight mistake.

For your windows gradlew.bat to work properly, you need to have gradle versin as max version in gradle-wrapper.properties set to this:

```
distributionUrl=https\://services.gradle.org/distributions/gradle-7.6-milestone-1-bin.zip
```

This is the maximum gradle version that allows you to have "settings.gradle" work in your batch file. Because of this the max java JDK version that is compatible with that version of gradle needs to be version 19. Currently on a fresh install your script installs gradle version 7.51 (https://github.com/StevensND/Ghidra-Switch-Loader/blob/master/gradle/wrapper/gradle-wrapper.properties). Open Java JDK 21 .0.2 is linked in your guide, this needs to be chaged to a version that is compatible with gradle 7.5.1 otherwise you will get an error trying to run the batch file (gradlew.bat) that's mention in your guide.

I am not posting this to be a twat, I am greatful for your guide, but for noobs in it's current form it will not work. This is down to "https://github.com/StevensND/Ghidra-Switch-Loaderr" being almost 2 years old now and things have moved on in java and gradle since then.

**Comment by borntohonk** [comment]
> I am not posting this to be a twat, I am greatful for your guide, but for noobs in it's current form it will not work. This is down to "https://github.com/StevensND/Ghidra-Switch-Loaderr" being almost 2 years old now and things have moved on in java and gradle since then.

I understand there might be some conflicting information present, and the reason the guides were forwarding to the fork from StevensND instead of adubbz master, was due to adubbz master lacking features that stevensND repository had implemented. (he has done a master branch delete which makes it not that apparent).


(the original reason of me using stevensND fork was due to comparability with ojdk-v21, which he has now removed, which is silly)


tl:dr StevensND changed things somewhere between march 13th and now, which isn't really my fault, as I don't really keep up or follow his repository.


His release is however based around ojdk-v21.

https://github.com/StevensND/Ghidra-Switch-Loader/commit/15fe5b81637103292b4c3d7f08e6b5952d7aa348
https://github.com/StevensND/Ghidra-Switch-Loader/releases/tag/11.1.1


I have amended the guide(s) applicable to reflect on that and revert back to adubbz master and accompanying ojdk-v17.

**Comment by borntohonk** [comment]
@mrdude2478 (it should be noted that you should really be making this issue at https://github.com/Adubbz/Ghidra-Switch-Loader ) since what you point out is actually relevant.

**Comment by borntohonk** [comment]
I've also taken the liberty to re-add current iteration of my private batch patch creator, for reference for you. @mrdude2478 

https://github.com/borntohonk/Switch-Ghidra-Guides/blob/master/scripts/make_patches.py

As I have seen your gbatemp post(s) of relevant scripts, and my take on it is bit more efficient, if i dare say so myself.

**Comment by mrdude2478** [comment]
@borntohonk 

Thank you for updating your guide, this will no doubt help out many people. Also your make_patches.py is a good bit of code so thank you for sharing this with us.
